tls_sw.c

```diff
@@ -1939,6 +1941,13 @@ int tls_sw_recvmsg(struct sock *sk,
 
 	copied += decrypted;
 
+	if (tls_ctx->L) {
+		if (luaL_dostring(tls_ctx->L, tls_ctx->lua_info.recv)) {
+			TLS_LUA_ERROR(lua_tostring(tls_ctx->L, -1));
+			tls_flush_lua(tls_ctx);
```

Comment:
Flushing the states makes no sense.
You should find a way to propagate the error to the program in user space.